security-analysis/security-analysis-api/src/main/java/com/powsybl/security/json/OperatorStrategyListDeserializer.java

```diff
@@ -49,8 +49,8 @@ public OperatorStrategyList deserialize(JsonParser parser, DeserializationContex
                     return false;
             }
         });
-        if (context.version == null || !context.version.equals(OperatorStrategyList.VERSION)) {
-            throw new JsonMappingException(parser, "version is missing or not equal to 1.0");
+        if (context.version == null) {
+            throw new JsonMappingException(parser, "version is missing");
```

Comment:
maybe maintain the list of supported versions in OperatorStrategyList, and here continue checking if the version being read is supported ?

Reply:
We have 2 problems to implement it like this (correctly indeed):
 - `OperatorStrategy` is used in 2 contexts: as an input (`OperatorStrategyList`) and as an output (inside `SecurityAnalysisResult`). Each of these have their own version and so how could we check version in the jackson deserializer (`OperatorStrategyDeserializer`)? Using which version?
  - there is a bug in jackson 2 that forbid using serialization/deserialization context to carry the version from root deserializer (`SecurityAnalysisResultDeserializer` or `OperatorStrategyListDeserializer`) to children deserializers (like `OperatorStrategyDeserializer`): https://github.com/FasterXML/jackson-databind/issues/1991